istioctl/cmd/proxyconfig.go

```diff
@@ -190,6 +303,94 @@ func proxyConfig() *cobra.Command {
 	listenerConfigCmd.PersistentFlags().StringVar(&listenerType, "type", "", "Filter listeners by type field")
 	listenerConfigCmd.PersistentFlags().IntVar(&port, "port", 0, "Filter listeners by Port field")
 
+	loggingCmd := &cobra.Command{
+		Use:   "logging <pod-name[.namespace]>",
+		Short: "(experimental) Retrieves logging levels of the Envoy in the specified pod",
+		Long:  "(experimental) Retrieve information about logging levels of the Envoy instance in the specified pod, and update optionally",
+		Example: `  # Retrieve information about logging levels for a given pod from Envoy.
+  istioctl proxy-config logging <pod-name[.namespace]>
+
+  # Update levels of the specified loggers and retrieve all the information about logging levels.
+  istioctl proxy-config logging <pod-name[.namespace]> --log_level=all:debug
+
+  # Reset levels of all the loggers to default value (warning) and retrieve all the information about logging levels.
+  istioctl proxy-config logging <pod-name[.namespace]> -r
+`,
+		Aliases: []string{"o"},
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				cmd.Println(cmd.UsageString())
+				return fmt.Errorf("logging requires pod name")
+			}
+			return nil
+		},
+		RunE: func(c *cobra.Command, args []string) error {
+			destLoggerLevels := map[string]Level{}
+			if reset {
+				// reset logging level to `defaultOutputLevel`, and ignore the `--log_level` option
+				destLoggerLevels[DefaultLoggerName] = defaultOutputLevel
+			} else if logLevelString != "" {
+				// parse `logLevelString` and update logging level of envoy
+				levels := strings.Split(logLevelString, ",")
+				for _, ol := range levels {
+					if !strings.Contains(ol, ":") || strings.Index(ol, "all:") == 0 {
+						level, ok := stringToLevel[ol[strings.Index(ol, ":")+1:]]
+						if ok {
+							destLoggerLevels = map[string]Level{
+								DefaultLoggerName: level,
+							}
+							break
+						}
+					} else {
+						loggerLevel := strings.Split(ol, ":")
+						if len(loggerLevel) != 2 {
+							break
+						}
+
+						index := sort.SearchStrings(activeLoggers, loggerLevel[0])
+						ok1 := index < len(activeLoggers) && activeLoggers[index] == loggerLevel[0]
+						level, ok2 := stringToLevel[loggerLevel[1]]
+						if !ok1 || !ok2 {
+							break
+						}
+						destLoggerLevels[loggerLevel[0]] = level
+					}
+				}
+			}
+
+			podName, ns := handlers.InferPodInfo(args[0], handlers.HandleNamespace(namespace, defaultNamespace))
+			var resp []byte
+			var err error
+			if len(destLoggerLevels) == 0 {
+				resp, err = setupLoggingEnvoyConfigWriter("", podName, ns)
+			} else {
+				for l, level := range destLoggerLevels {
+					resp, err = setupLoggingEnvoyConfigWriter(l+"="+levelToString[level], podName, ns)
+				}
+			}
+			if err != nil {
+				return err
+			}
+			_, _ = fmt.Fprint(c.OutOrStdout(), string(resp))
```

Comment:
Not sure if its a good idea or not, what do you guys think of outputting something like "view logs with `kubectl logs POD -c istio-proxy`" at the bottom?

Reply:
Possibly, but it would also be easy to add later if we need it.

Reply:
I agree with @ayj.